src/_zkapauthorizer/tests/test_plugin.py

```diff
@@ -190,14 +194,20 @@ def _replication_enabled_connection_test(self, now: datetime, enabled: bool):
         # Create the underlying database file.
         store = open_store(lambda: now, memory_connect, config)
 
-        self.assertThat(
-            # Right now enabling replication does absolutely nothing except
-            # flip this flag, so there's no other behavior to observe in this
-            # test.  Maybe later when there's more replication implementation,
-            # we can assert something more meaningful.
-            store._connection._replicating,
-            Equals(enabled),
-        )
+        if isinstance(store._connection, _ReplicationCapableConnection):
```

Comment:
if we check the type of `store._connection` then mypy allows us to touch attributes of the concrete type.